libraries/AP_Mission/AP_Mission_Commands.cpp

```diff
@@ -165,6 +165,17 @@ bool AP_Mission::start_command_camera(const AP_Mission::Mission_Command& cmd)
         }
         return false;
 
+    case MAV_CMD_SET_CAMERA_SOURCE:
```

Comment:
I'd like to see
```suggestion
#if AP_MAVLINK_MAV_CMD_SET_CAMERA_SOURCE_ENABLED
    case MAV_CMD_SET_CAMERA_SOURCE:
```
there are similar defines already for other commands.

Reply:
Hi, so is this an automatically generated definition or is this something I should add?  I think we're getting too detailed if we're adding new definitions for individual mavlink commands.

Reply:
No, we need to have a pattern so that we can continue to compile on F405 boards etc.

Reply:
As discussed separately this feature apparently costs about 550 bytes on the F405 boards.  I really don't think that removing part of a feature by introducing an additional define which then ends up in the custom build server introducing even more complexity to the user is the way to go.  If we're concerned about space on the F405 boards then let's find something bigger to remove.  A driver, a large feature, a flight mode.. these all make sense but this little feature is too small to be worth the additional complexity.

Reply:
> As discussed separately this feature apparently costs about 550 bytes on the F405 boards. I really don't think that removing part of a feature by introducing an additional define which then ends up in the custom build server introducing even more complexity to the user is the way to go. If we're concerned about space on the F405 boards then let's find something bigger to remove. A driver, a large feature, a flight mode.. these all make sense but this little feature is too small to be worth the additional complexity.

The new define would not appear on the custom build server; it would simply get dropped into `minimize_features.inc`.

We don't want to add this dead code which then costs us an actual feature on these boards at some stage.

Reply:
OK, i've added HAL_MOUNT_SET_CAMERA_SOURCE enabled definition that removes the mount code if we don't have a gimbal enabled that may have multiple lenses.  There is more code in AP_Camera and AP_Mission but its relatively thin boiler plate code so I think this is really diminishing returns and would add more dependencies on AP_Mount.